> Failing on darwin with:
> 
> ```
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
>   File "<string>", line 1, in <lambda>
>   File "/nix/store/8zc3wcplydp8gsxms24scpzdca438dk5-python3-3.12.8/lib/python3.12/importlib/__init__.py", line 90, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 999, in exec_module
>   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
>   File "/nix/store/z21xzks0wrcxmz6xgx1pijzqiadi1k69-python3.12-biosppy-2.2.2/lib/python3.12/site-packages/biosppy/inter_plotting/__init__.py", line 18, in <module>
>     from . import ecg, acc
>   File "/nix/store/z21xzks0wrcxmz6xgx1pijzqiadi1k69-python3.12-biosppy-2.2.2/lib/python3.12/site-packages/biosppy/inter_plotting/ecg.py", line 20, in <module>
>     from tkinter import *
>   File "/nix/store/8zc3wcplydp8gsxms24scpzdca438dk5-python3-3.12.8/lib/python3.12/tkinter/__init__.py", line 38, in <module>
>     import _tkinter # If this fails your Python may not be configured for Tk
>     ^^^^^^^^^^^^^^^
> ModuleNotFoundError: No module named '_tkinter'
> ```

I could add it as an optional dependency on darwin builds but I don't have a machine to test for other failures

--- a/pkgs/development/python-modules/biosppy/default.nix
+++ b/pkgs/development/python-modules/biosppy/default.nix
@@ -1,0 +1,70 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  buildPythonPackage,
+  setuptools,
+  bidict,
+  h5py,
+  matplotlib,
+  numpy,
+  scikit-learn,
+  scipy,
+  shortuuid,
+  six,
+  joblib,
+  pywavelets,
+  mock,
+  tkinter,
+  opencv-python,
+}:
+
+buildPythonPackage rec {
+  pname = "biosppy";
+  version = "2.2.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "scientisst";
+    repo = "BioSPPy";
+    tag = "v${version}";
+    hash = "sha256-U0ZftAlRlazSO66raH74o/6eP1RpmuFoA6HJ+xmgKR8=";
+  };
+
+  build-system = [
+    setuptools
+  ];
+
+  dependencies = [
+    opencv-python
+    bidict
+    h5py
+    matplotlib
+    numpy
+    scikit-learn
+    scipy
+    shortuuid
+    six
+    joblib
+    pywavelets
+    mock
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [ tkinter ];
+
+  doCheck = false; # no tests
+
+  pythonImportsCheck = [
+    "biosppy"
+    "biosppy.signals"
+    "biosppy.synthesizers"
+    "biosppy.inter_plotting"
+    "biosppy.features"
+  ];
+
+  meta = {
+    description = "Biosignal Processing in Python";
+    homepage = "https://biosppy.readthedocs.io/";
+    changelog = "https://github.com/scientisst/BioSPPy/releases/tag/v${version}";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ genga898 ];
+  };
+}

--- a/pkgs/development/python-modules/neurokit2/default.nix
+++ b/pkgs/development/python-modules/neurokit2/default.nix
@@ -1,0 +1,101 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  buildPythonPackage,
+  setuptools,
+  pytest,
+  scipy,
+  scikit-learn,
+  pandas,
+  matplotlib,
+  requests,
+  cvxopt,
+  biosppy,
+  pytest-cov-stub,
+  mock,
+  plotly,
+  astropy,
+  coverage,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "neurokit2";
+  version = "0.2.10";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "neuropsychology";
+    repo = "NeuroKit";
+    tag = "v${version}";
+    hash = "sha256-e/B1JvO6uYZ6iVskFvxZLSSXi0cPep9bBZ0JXZTVS28=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace-fail '"pytest-runner"' '"pytest"'
+  '';
+
+  build-system = [
+    setuptools
+    pytest
+  ];
+
+  dependencies = [
+    scipy
+    scikit-learn
+    pandas
+    matplotlib
+    requests
+    cvxopt
+    biosppy
+  ];
+
+  nativeCheckInputs = [
+    pytest-cov-stub
+    mock
+    plotly
+    astropy
+    coverage
+    pytestCheckHook
+  ];
+
+  disabledTests = lib.optionals stdenv.hostPlatform.isDarwin [
+    # Crash in matplotlib (Fatal Python error: Aborted)
+    "test_events_plot"
+  ];
+
+  disabledTestPaths = [
+    # Required dependencies not available in nixpkgs
+    "tests/tests_complexity.py"
+    "tests/tests_eeg.py"
+    "tests/tests_eog.py"
+    "tests/tests_ecg.py"
+    "tests/tests_bio.py"
+    "tests/tests_data.py"
+    "tests/tests_epochs.py"
+    "tests/tests_ecg_findpeaks.py"
+    "tests/tests_eda.py"
+    "tests/tests_emg.py"
+    "tests/tests_hrv.py"
+    "tests/tests_rsp.py"
+    "tests/tests_ppg.py"
+    "tests/tests_signal.py"
+
+    # Dependency is broken `mne-python`
+    "tests/tests_microstates.py"
+  ];
+
+  pythonImportsCheck = [
+    "neurokit2"
+  ];
+
+  meta = {
+    description = "Python Toolbox for Neurophysiological Signal Processing";
+    homepage = "https://github.com/neuropsychology/NeuroKit";
+    changelog = "https://github.com/neuropsychology/NeuroKit/releases/tag/v${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ genga898 ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -1677,6 +1677,8 @@ self: super: with self; {
 
   biopython = callPackage ../development/python-modules/biopython { };
 
+  biosppy = callPackage ../development/python-modules/biosppy { };
+
   biothings-client = callPackage ../development/python-modules/biothings-client { };
 
   biplist = callPackage ../development/python-modules/biplist { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -9191,6 +9191,8 @@ self: super: with self; {
 
   neuralfoil = callPackage ../development/python-modules/neuralfoil { };
 
+  neurokit2 = callPackage ../development/python-modules/neurokit2 { };
+
   neuron-full = pkgs.neuron-full.override { python3 = python; };
 
   neuronpy = toPythonModule neuron-full;

